base_layer/wallet_ffi/src/lib.rs

```diff
@@ -1681,7 +1689,20 @@ pub unsafe extern "C" fn transport_tor_create(
     let mut identity = None;
     if !tor_identity.is_null() {
         let bytes = (*tor_identity).0.as_slice();
-        identity = Some(Box::new(tor::TorIdentity::from_binary(bytes).unwrap()));
+        match tor::TorIdentity::from_binary(bytes) {
+            Ok(ident) => {
+                identity = Some(Box::new(ident));
+            },
+            Err(err) => {
+                error = LibWalletError::from(InterfaceError::DeserializationError(format!(
```

Comment:
What error is returned here?
Could it be useful to log the actual error here before changing the error to the new one?

Reply:
It's included in the string in `DeserializationError`, not sure if that is logged I'm keeping with the pattern used elsewhere in wallet ffi - @StriderDM - would a log here be beneficial on mobile or is it already logged somewhere?